Docs 6745 export mule app mc - download complete mule app (#125)
mulesoft/docs-design-center: modules/ROOT/pages/fd-deploy-app-from-rm.adoc

= Deploy a Mule Application from Runtime Manager

It might be the case that you are not allowed to deploy an application directly from Flow Designer.
Your organization might require someone with the proper authority to deploy applications from Runtime Manager.

> You can share with that person a copy of your project as a `.jar` file to deploy your application with.

Reviewer: Suggest us saying "In that scenario Flow Designer allows you to export the project as a `.jar` file and share it with those who are responsible for app deployment"

Author: Rewritten to accomodate.

== Procedure

. In Flow Designer, follow these steps:
.. Ensure that your project is free of errors.
.. Export your project by clicking the dots next to the Test button and select *Export*.

> .. Select *Export project sources* to export just your project source files, or

Reviewer: @mcherem why in this section of the documentation would we even talk about the **Export project sources** option, especially when we don't provide additional guidance to the user on when to use which option. i suggest we kill this line.

Reviewer: or if this is the only place where we talk about the Export feature, then we need to explain in greater detail why someone would want to select this option. I would like to ask that you add
- **what it is for** (ie. when a user wants to export the project for the purposes of Importing it to Studio to take advantage of some of the advanced features available there, or if they want to collaborate with another someone else working on a Flow Designer project, and want to make it available for importing it there) 
- **what is the output** (i.e. what we exclude/include in this option vs the other option)

Author: Rewritten to reflect comments.

> *Export mule application* to export your entire Mule app, then click *Export*.

Reviewer: @mcherem suggest we are more explicit and guide them to use this option when trying to share a file to be used for deployment in Runtime manager... "Select *Export mule application* (Selecting this option will produce a `.jar` file which will include all the connector credentials (@mcoaker to confirm) as well as all artifacts needed by Runtime Manager) 

something like that

Author: Rewritten to reflect comments.

+

> *Result:* A `.jar` file that contains a copy of your project source or Mule app is downloaded to the default location on your computer.

Reviewer 2: i'm not sure about this part. If we need to be more specific about the differences between those two, or if the information inside the modal itself is enough. Maybe wait for @mcoaker review

The rest in my opinion is ok :)

Author: Fixed.

> = Deploy a Mule Application from Runtime Manager

Reviewer 3: I wonder if this title is really accurate.
This task is mostly about sending the app .jar file to someone who can, in Runtime Manager, deploy the app to CloudHub.
It's kind of confusing.

> = Deploy a Mule Application from Runtime Manager

Author: You are correct, the process is very confusing. There are scenarios where certain types of FD users (usually the Designers) are not allowed to deploy Mule apps into production. In this case, they will need to 'give' the file to some else to do that for them. That someone else will likely deploy it in Runtime Manager. The sad truth is that the user needs to export the .jar file and get it over to the other user to deploy. Not a good user experience. I think the title reflects what the user wants to do, it's just that it's a clunky way of doing it.

> = Deploy a Mule Application from Runtime Manager

Author: Fixed.

> Perhaps your organization requires someone with the proper authority to deploy applications from Mule.

Reviewer: 'deploy applications from Mule.' suggest change to 'deploy Mule applications once the design phase has been completed'

Author: Fixed.

> Perhaps your organization requires someone with the proper authority to deploy Mule applications once the design phase has been completed.

Author: The way I understand it, there are different personas using the product. One persona is a designer type that designs apps all day but is not the one who deploys them into production. In this case, one persona hands off the app to another. The way the PM describes it, this is the case where you need to give the app to someone who can deploy it. The other file covers the other use case. I can reword it slightly, but the reason is not a limit with the tool, it's a limit by persona.
Here's a rewrite with your feedback in mind :
After designing a Mule application in Anypoint Flow Designer, someone other than yourself may need to deploy it. You must first use Flow Designer to export the project as a .jar file, in order to share it with someone with deployment credentials so that they can deploy the app using Anypoint Runtime Manager.

> Perhaps your organization requires someone with the proper authority to deploy Mule applications once the design phase has been completed.

Author: Fixed.

Flow Designer allows you to export the project as a `.jar` file and share it with those who are responsible for Mule app deployment.
You'll need to use *Export mule application*, as it produces a `.jar` file that includes all the connector credentials and artifacts required by the Mule runtime.

> . In Flow Designer, ensure that your project is free of errors.

Reviewer 3: This is a prerequisite to the deployment task, not the first step.

Consider identifying it as such, using a title, or working it into the intro text: "After designing a Mule application in Anypoint Flow Designer and ensuring that it is free of errors...."

Author: Fixed.

> . Export your project by clicking the dots next to the Test button and select *Export* from the list.

Reviewer 3: Is it a bit odd that Export is part of Test?
Do you think we should show the UI?

Author: I was just giving orientation for the three dots (...). They represent a drop-down list and are next to the Test button. It makes sense because first you test, then deploy or export.

> . Select *Export mule application* for the entire app, and click *Export*.

Reviewer 3: I don't understand what you mean by selecting "for the entire app"? [Ah, now I do. How about "...to export the entire app, and then ..."]
Do we need a screen capture?
Would it suffice to say, following the previous step to select Export, "Select Export mule application and click Export"?

Author: You've identified an important point. I think I need to explain that there is a pop-up with a choice it will make more sense.

Author: Fixed.

+
A `.jar` file containing a copy of your Mule app downloads to the default location on your computer.

> . Place the `.jar` file in a location that the other person can access.

Reviewer 3: ```suggestion
. Move the `.jar` file to a location that your intended recipient can access.
```

Author: Fixed.

> . Have the person receiving the `.jar` file follow these steps in Runtime Manager:

Reviewer 3: I'm trying to think of a way that avoids "Have the person...do this:"
Maybe the step is really 

"4. Alert the recipient that the deployment .jar is available so that they can:
      a. Access the .jar file.
      b. Change environments to that in which....   [Does this mean they have to move the .jar there?]
      c. Click....

Author: The way it works is that from RM they follow the steps for deploying. They will be given a selection that allows them to upload the .jar file. So, we don't want to rewrite the RM deploy steps - just give them the info on the jar file.

Author: Fixed.
